packages/worker/src/debug-replayer/workflow-notifier.ts

```diff
@@ -0,0 +1,40 @@
+import { workflowInfo } from '@temporalio/workflow';
+
+class WorkflowNotifier {
+  static _instance?: WorkflowNotifier;
+
+  /**
+   * Access the singleton instance - one per workflow context
+   */
+  static instance(): WorkflowNotifier {
+    if (this._instance === undefined) {
+      this._instance = new this();
+    }
+    return this._instance;
+  }
+
+  private constructor() {}
+
+  lastNotifiedStartEvent = -1;
+
+  notifyRunner(): void {
+    const eventId = workflowInfo().historyLength;
+    if (this.lastNotifiedStartEvent >= eventId) return;
```

Comment:
Why this condition? Do you mean to avoid notifying multiple times during per workflow task?

Assuming you are, why call `notifyRunner()` from outbound-interceptor?

Reply:
This condition is meant to avoid notifying for every operation.
We call from outbound interceptor after an operation is resolved (activity completes for example), that might be the first event to process in a workflow task.

Reply:
Ok, it's clear now. I had missed the fact that in outgoing, the notification would be sent only _after_ the completion of a command promise. Clever.